src/controllers/controller.bar.js

```diff
@@ -383,11 +388,16 @@ module.exports = function(Chart) {
 			var meta = me.getMeta();
 			var scale = me.getValueScale();
 			var datasets = chart.data.datasets;
-			var value = scale.getRightValue(datasets[datasetIndex].data[index]);
+			// float-bar support, if y arguments are array function will use top - bottom value to calculate bar height
+			var value = scale.parseValue(datasets[datasetIndex].data[index]);
 			var stacked = scale.options.stacked;
 			var stack = meta.stack;
+			// float-bar support, if y arguments are array function will use proper value as bar start point
 			var start = 0;
-			var i, imeta, ivalue, base, head, size;
+			var i, imeta, ivalue, base, head, size, yStackValue;
+			if (value.min !== value.max) {
+				start = value.val >= 0 ? value.min : value.max;
```

Comment:
`val` is defined as `max - min`. `max` is always greater than `min` here. That means that `value.val >= 0` will always be true.  So you could just make this line `start = value.min`

Reply:
max is always greater than min, thats true. But, value.val >= 0 only for positive numbers like [1,4], and if we have [-1,-4] in this case value.val will be -3 and its < 0. So, for start we are using max or min value that depends on the positive or negative value. Its because we are ploting bars relatively the zero line, and in case of positive number we should build next stacked bar from the top of previous bar (from max value), for example [1,4], we will plot from 4, because 4 > 1. But in the case of negative number we are plotting this value in opposite site because -1 > -4. So, the top of negative values will be low value in the pair. Its like mirror effect, when your mirror is actually 0 line. So, it should be like it is here.